tests/plugins/test_pin_operator_digests.py

```diff
@@ -130,16 +132,20 @@ def mock_operator_csv(tmpdir, filename, pullspecs):
     return path
 
 
-def mock_package_mapping_files(tmpdir, repo_replacements):
+def mock_package_mapping_files(repo_replacements):
     repo_replacements = repo_replacements or {}
 
-    # write mappings to files, update repo_replacements to point to those files
+    # create unique url for each registry, mock responses, update mapping to point to urls
     for registry, mapping in repo_replacements.items():
-        filename = 'mapping-{}.yaml'.format(registry)
-        path = tmpdir.join(filename)
-        with open(str(path), 'w') as f:
-            yaml.dump(mapping, f)
-        repo_replacements[registry] = str(path)
+        url = 'https://somewhere.net/mapping-{}.yaml'.format(registry)
+
+        # ruamel.yaml does not support dumping to str, use an io stream
+        f = io.BytesIO()
```

Comment:
Why BytesIO and not StringIO?

Reply:
Originally because the `iter_content()` of a response is bytes, but turns out you can make it unicode with the `decode_unicode` param